boat-maven-plugin/src/main/java/com/backbase/oss/boat/RadioMojo.java

```diff
@@ -0,0 +1,171 @@
+package com.backbase.oss.boat;
+
+import com.backbase.oss.boat.bay.client.ApiClient;
+import com.backbase.oss.boat.bay.client.api.BoatMavenPluginApi;
+import com.backbase.oss.boat.bay.client.model.BoatLintReport;
+import com.backbase.oss.boat.bay.client.model.UploadRequestBody;
+import com.backbase.oss.boat.bay.client.model.UploadSpec;
+import com.backbase.oss.boat.loader.OpenAPILoader;
+import com.backbase.oss.boat.loader.OpenAPILoaderException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationConfig;
+import feign.auth.BasicAuthRequestInterceptor;
+import io.swagger.v3.oas.models.OpenAPI;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(name = "radio", requiresDependencyResolution = ResolutionScope.RUNTIME, threadSafe = true)
+@Slf4j
+@Getter
+@Setter
+public class RadioMojo extends AbstractMojo {
+
+    @Parameter(property = "groupId", defaultValue = "${project.groupId}", required = true)
+    private String groupId;
+
+    @Parameter(property = "artifactId", defaultValue = "${project.artifactId}", required = true)
+    private String artifactId;
+
+    @Parameter(property = "version", defaultValue = "${project.version}", required = true)
+    private String version;
+
+    @Parameter(property = "basePath", required = true)
+    private String basePath;
```

Comment:
What is a typical basePath value? It seems to be an odd required parameter.

Reply:
Its the basepath of the boat-bay server. Updated to `boatBayUrl` and added java doc.